logpyle/__init__.py

```diff
@@ -638,18 +634,72 @@ def __init__(self, filename: Optional[str] = None, mode: str = "r",
 
         # {{{ warnings/logging capture
 
-        self.warning_data: List[_LogWarningInfo] = []
+        warning_data: List[_LogWarningInfo] = []
+        self.warning_data = warning_data
         self.old_showwarning: Optional[Callable[..., Any]] = None
         if capture_warnings and self.mode[0] == "w":
             self.capture_warnings(True)
 
-        self.logging_data: List[_LogWarningInfo] = []
+        logging_data: List[_LogWarningInfo] = []
+        self.logging_data = logging_data
         self.logging_handler: Optional[logging.Handler] = None
         if capture_logging and self.mode[0] == "w":
             self.capture_logging(True)
 
         # }}}
 
+        # {{{ atexit handling
+
+        def atexit_close() -> None:
+            # Note: This function can not reference 'self', otherwise
+            # atexit will hold a reference to the LogManager instance,
+            # keeping that instance alive.
```

Comment:
Why is that so bad? `LogManager`s, like files, should be explicitly closed by the user.

Reply:
The only issue is that the LogManager can not be deleted unless it is closed first. If that is ok, there is no need for this code.

Reply:
c4ec70a changes this to not have a separate atexit function anymore, with the caveat above.